src/Microsoft.IdentityModel.Tokens/AsymmetricSignatureProvider.cs

```diff
@@ -181,7 +177,7 @@ protected virtual HashAlgorithmName GetHashAlgorithmName(string algorithm)
             throw LogHelper.LogException<ArgumentOutOfRangeException>(LogMessages.IDX10640, algorithm);
         }
 
-        private void ResolveDotNetCoreAsymmetricAlgorithm(AsymmetricSecurityKey key, string algorithm, bool willCreateSignatures)
+        public virtual void ResolveAsymmetricAlgorithm(AsymmetricSecurityKey key, string algorithm, bool willCreateSignatures)
```

Comment:
Hmm, this does not look right and it makes me think if we even need to make this virtual. There are multiple ways to allow users to provide their own Rsa/Ecdsa implementation: 
1. SecurityKey constructor
2. SecurityKey.GetSignatureProvider override
3. CryptoProviderFactory.CreateForSigning/Verfiying override

All of them will impact the custom implementation/override at different level. CryptoProviderFactory having the most broader effect.

Also, there does not seem to be a clean way to provide an override at this level. All fields are private and we cannot have this method return `AsymmetricAlgorithm` type since that is too vanilla. I am thinking its best to keep it as it was before.
@brentschmaltz thoughts?